llvm/test/DebugInfo/CAS/AArch64/debug_line_dedupe.test

```diff
@@ -0,0 +1,96 @@
+RUN: rm -rf %t && mkdir -p %t
+RUN: split-file %s %t
+
+RUN: llc -O0 --filetype=obj --cas-backend --cas=%t/cas --mccas-casid %t/a.ll -o %t/a.id
+RUN: llc -O0 --filetype=obj --cas-backend --cas=%t/cas --mccas-casid %t/b.ll -o %t/b.id
+RUN: llvm-cas-dump --cas=%t/cas --casid-file %t/a.id %t/b.id | FileCheck %s
+
+CHECK: mc:debug_line   llvmcas://[[CASID:[a-z0-9]+]]
+CHECK-NEXT: mc:debug_line   llvmcas://
+
+CHECK: mc:debug_line   llvmcas:// 
+CHECK-NEXT: mc:debug_line   llvmcas://[[CASID]]
+CHECK-NEXT: mc:debug_line   llvmcas:// 
+
+//--- a.ll
+
+target triple = "arm64-apple-macosx12.0.0"
+define void @foo() #0 !dbg !9 {
+  ret void, !dbg !14
+}
+define i32 @bar(i32 noundef %x) #0 !dbg !15 {
+entry:
+  %x.addr = alloca i32, align 4
+  %0 = load i32, ptr %x.addr, align 4, !dbg !21
+  %add = add nsw i32 %0, 1, !dbg !22
+  ret i32 %add, !dbg !23
+}
```

Comment:
Do we need this many instructions? I suspect we can make the debug metadata a lot more minimal if we clean up this function a bit more.

Related: what would happen if both `foo` and `bar` had the _exact_ same body, for example `ret void`, with the same  debug information in both functions? Can we test deduplication with the below, without the need for two "IR files"?

I understand that here you're trying to mimic the C behaviour, but maybe we can get away with a much simpler test at the IR level

```
define void @foo() #0 !dbg !9 {
   ret void, !dbg !14
 }
define void @bar() #0 !dbg !9 {
   ret void, !dbg !14
 }
```